### PR TITLE
fix: Granular sync fixes

### PR DESCRIPTION
**Pull request by Balcan**
- Sync button only appears when status is not synced
- SMS sync can't show in home screen elements
- Fixed granular sync dialog state when there was one completed dataSetInstance
- Prevent crash if enrollment list is empty

[issue](https://jira.dhis2.org/browse/ANDROAPP-2612)

### CODE REVIEW

**Comment by JaimeToca** [comment]
Can I have some unit test @Balcan :dancer:

**Comment by Balcan** [comment]
> Can I have some unit test @Balcan 💃

NO! :P
ok...

File: app/src/test/java/org/dhis2/utils/granularsync/GranularSyncPresenterTest.kt [hunk 1]
```diff
@@ -113,6 +117,133 @@ class GranularSyncPresenterTest {
...
+            d2.dataSetModule().dataSetCompleteRegistrations()
+        ) doReturn mock()
+        whenever(
+            d2.dataSetModule().dataSetCompleteRegistrations()
+                .byDataSetUid()
+        ) doReturn mock()
+        whenever(
+            d2.dataSetModule().dataSetCompleteRegistrations()
+                .byDataSetUid().eq("data_set_uid")
+        ) doReturn mock()
```
Comment (MarinaCape) [inline app/src/test/java/org/dhis2/utils/granularsync/GranularSyncPresenterTest.kt]: You can just join that lines using d2.dataSetModule().dataSetCompleteRegistrations()
                .byDataSetUid().eq("data_set_uid") in first time

**Comment by ferdyrod** [comment]
Tested on device!

**Pull request merged**